src/app/builders/components/Table/BuilderHeaderRow.tsx

```diff
@@ -0,0 +1,210 @@
+'use client'
+
+import { Column, Sort, TableAction, useTableActionsContext, useTableContext } from '@/shared/context'
+import { HtmlHTMLAttributes, Suspense } from 'react'
+
+import { CommonComponentProps } from '@/components/commonProps'
+import { ArrowDownWFill } from '@/components/Icons/v3design/ArrowDownWFill'
+import { ArrowsUpDown } from '@/components/Icons/v3design/ArrowsUpDown'
+import { ArrowUpWFill } from '@/components/Icons/v3design/ArrowUpWFill'
+import { TableHeaderCell, TableHeaderNode } from '@/components/TableNew'
+import { Label, Paragraph } from '@/components/TypographyNew'
+import { cn } from '@/lib/utils'
+import { Dispatch, FC } from 'react'
+import { SelectorHeaderCell } from './Cell/SelectorHeaderCell/SelectorHeaderCell'
+
+// Column types and ids definitions
+const COLUMN_IDS = [
+  'builder',
+  'backing',
+  'backer_rewards',
+  'rewards_past_cycle',
+  'rewards_upcoming',
+  'allocations',
+  'actions',
+] as const
+export type ColumnId = (typeof COLUMN_IDS)[number]
+export const isColumnId = (id: string): id is ColumnId => COLUMN_IDS.includes(id as ColumnId)
+
+// FIXME: Leave to the end to fix the dynamic width of the columns so that the actions column expands when other columns are hidden without affecting the builder column.
+export const COLUMN_WIDTHS: Record<ColumnId, HtmlHTMLAttributes<HTMLTableCellElement>['className']> = {
+  builder: 'grow-3 shrink-2 basis-6 pl-4',
+  backing: 'grow-1 shrink-1 basis-6 pl-4',
+  backer_rewards: 'grow-1 shrink-1 basis-6 pl-4',
+  rewards_past_cycle: 'grow-1 shrink-1 basis-6 pl-4',
+  rewards_upcoming: 'grow-1 shrink-1 basis-6 pl-4',
+  allocations: 'grow-1 shrink-0 basis-6 pl-4',
+  actions: 'grow-1 shrink-0 basis-6 pl-4',
+}
+const OrderIndicatorContainer: FC<CommonComponentProps> = ({ className, children }) => (
+  <div className={cn('flex pt-1 justify-center gap-2', className)}>{children}</div>
+)
+
+const OrderIndicator: FC<CommonComponentProps & { column: Column['id'] }> = ({ column }) => {
+  const { sort } = useTableContext<ColumnId>()
+
+  if (!sort || column === undefined) return null
+
+  if (sort.by !== column) {
+    return (
+      <OrderIndicatorContainer>
+        <ArrowsUpDown />
+      </OrderIndicatorContainer>
+    )
+  }
+
+  if (sort.direction === 'asc') {
+    return (
+      <OrderIndicatorContainer>
+        <ArrowUpWFill />
+      </OrderIndicatorContainer>
+    )
+  }
+
+  return (
+    <OrderIndicatorContainer>
+      <ArrowDownWFill />
+    </OrderIndicatorContainer>
+  )
+}
+
+/**
+ * Dispatch a round robin based sort.
+ * If the column is not sorted, sort it descending.
+ * If the column is sorted and descending, sort it ascending.
+ * If the column is sorted and ascending, stop sorting.
+ * @param dispatch - The dispatch function to dispatch the sort action.
+ * @param columnId - The column id to sort.
+ * @param currentSort - The current sort state.
+ */
+const dispatchSortRoundRobin = (
```

Comment:
Not sure about the way it works. I'd expect something easier:
1. if not sorted, sort it ascending (or descending, not a big preference for the initial sort)
2. if sorted ascending, sort it descending
3. if sorted descending, sort it ascending.

I don't get the relationship with "round robin" either 😅

Reply:
BTW in the storybook, the behaviour is exactly what I'd expect, so probably it's me not understanding the comment 😅